internal/storage/v2/clickhouse/tracestore/dbmodel/to_dbmodel.go

```diff
@@ -0,0 +1,555 @@
+	return TraceColumnPair{
+		ColName: colName,
+		Col:     col,
+	}
+}
+
+func newAttributeColumns(acm *attributeColumnsMap, attributeType AttributeType) {
+	if attributeType == AttributeTypeEvent || attributeType == AttributeTypeLink {
+		acm.buildAttrColumns(attributeType, ValueTypeBool,
+			proto.NewArray(new(proto.ColStr).LowCardinality().Array()),
+			proto.NewArray(new(proto.ColBool).Array()))
+		acm.buildAttrColumns(attributeType, ValueTypeDouble,
+			proto.NewArray(new(proto.ColStr).LowCardinality().Array()),
+			proto.NewArray(new(proto.ColFloat64).Array()))
+		acm.buildAttrColumns(attributeType, ValueTypeInt,
+			proto.NewArray(new(proto.ColStr).LowCardinality().Array()),
+			proto.NewArray(new(proto.ColInt64).Array()))
+		acm.buildAttrColumns(attributeType, ValueTypeStr,
+			proto.NewArray(new(proto.ColStr).LowCardinality().Array()),
+			proto.NewArray(new(proto.ColStr).LowCardinality().Array()))
+		acm.buildAttrColumns(attributeType, ValueTypeBytes,
+			proto.NewArray(new(proto.ColStr).LowCardinality().Array()),
+			proto.NewArray(new(proto.ColStr).LowCardinality().Array()))
+	} else {
+		acm.buildAttrColumns(attributeType, ValueTypeBool, new(proto.ColStr).LowCardinality().Array(), new(proto.ColBool).Array())
+		acm.buildAttrColumns(attributeType, ValueTypeDouble, new(proto.ColStr).LowCardinality().Array(), new(proto.ColFloat64).Array())
+		acm.buildAttrColumns(attributeType, ValueTypeInt, new(proto.ColStr).LowCardinality().Array(), new(proto.ColInt64).Array())
+		acm.buildAttrColumns(attributeType, ValueTypeStr, new(proto.ColStr).LowCardinality().Array(), new(proto.ColStr).LowCardinality().Array())
+		acm.buildAttrColumns(attributeType, ValueTypeBytes, new(proto.ColStr).LowCardinality().Array(), new(proto.ColStr).LowCardinality().Array())
+	}
+}
+
+func (acm attributeColumnsMap) buildAttrColumns(attributeType AttributeType, valueType pcommon.ValueType, keyCol proto.Column, valueCol proto.Column) {
+	acm[valueType] = AttributeColumnPair{
+		keyColName:   attributeType.String() + "Attributes" + valueType.String() + "Key",
+		keyCol:       keyCol,
+		valueColName: attributeType.String() + "Attributes" + valueType.String() + "Value",
+		valueCol:     valueCol,
+	}
+}
+
+type AttributeType int32
+
+const (
+	AttributeTypeResource AttributeType = iota
+	AttributeTypeScope
+	AttributeTypeSpan
+	AttributeTypeEvent
+	AttributeTypeLink
+)
+
+func (at AttributeType) String() string {
+	switch at {
+	case AttributeTypeResource:
+		return "Resource"
+	case AttributeTypeScope:
+		return "Scope"
+	case AttributeTypeSpan:
+		return "Span"
+	case AttributeTypeEvent:
+		return "Event"
+	case AttributeTypeLink:
+		return "Link"
+	default:
+		return "Unknown"
+	}
+}
+
+// ToDBModel Converts the OTel pipeline Traces into a ClickHouse-compatible format for batch insertion.
+// It maps the trace attributes, spans, links and events from the OTel model to the appropriate ClickHouse column types
+func ToDBModel(td ptrace.Traces) proto.Input {
```

Comment:
please move to the top of the file

If `ToDBModel()` returns `proto.Input`, what value does dbmodel.go provide?